test/org/impact2585/frc2016/tests/WheelSystemTest.java

```diff
@@ -139,6 +144,25 @@ public void test() {
 		drivetrain.setCurrentRampForward(0.5);
 		drivetrain.run();
 		Assert.assertTrue(currentRampForward == ramp && rotate == 0);
+		
+		//tests rotation exponent
+		toggleRotationExponent = true;
+		rotate = 0.4;
+		drivetrain.run();
+		Assert.assertTrue(currentRampForward == ramp && rotate == Math.pow(0.4, WheelSystem.SECONDARY_ROTATION_EXPONENT));
+		
+		//test if the rotation exponent doesn't toggle if the toggle is pressed
+		toggleRotationExponent = true;
+		rotate = 0.4;
+		drivetrain.run();
+		Assert.assertTrue(currentRampForward == ramp && rotate == Math.pow(0.4, WheelSystem.SECONDARY_ROTATION_EXPONENT));
+		
+		//tests if the rotation exponent can be toggled off
+		toggleRotationExponent = false;
+		drivetrain.run();
+		rotate = 0.5;
+		toggleRotationExponent = true;
+		Assert.assertTrue(currentRampForward == ramp && rotate == Math.pow(0.5, WheelSystem.PRIMARY_ROTATION_EXPONENT));
```

Comment:
I did call drivetrain.run()

On Fri, May 20, 2016 at 2:26 PM, Michael Murphey <notifications@github.com>
wrote:

> In test/org/impact2585/frc2016/tests/WheelSystemTest.java
> <https://github.com/Impact2585/FRC2016/pull/99#discussion_r64094645>:
>
> > +		rotate = 0.4;
> > +		drivetrain.run();
> > +		Assert.assertTrue(currentRampForward == ramp && rotate == Math.pow(0.4, WheelSystem.SECONDARY_ROTATION_EXPONENT));
> > +		
> > +		//test if the rotation exponent doesn't toggle if the toggle is pressed
> > +		toggleRotationExponent = true;
> > +		rotate = 0.4;
> > +		drivetrain.run();
> > +		Assert.assertTrue(currentRampForward == ramp && rotate == Math.pow(0.4, WheelSystem.SECONDARY_ROTATION_EXPONENT));
> > +		
> > +		//tests if the rotation exponent can be toggled off
> > +		toggleRotationExponent = false;
> > +		drivetrain.run();
> > +		rotate = 0.5;
> > +		toggleRotationExponent = true;
> > +		Assert.assertTrue(currentRampForward == ramp && rotate == Math.pow(0.5, WheelSystem.PRIMARY_ROTATION_EXPONENT));
>
> It looks like you forgot to call drivetrain.run(). Also, the fact that
> the unit test is still passing with that seems like a bad sign.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Impact2585/FRC2016/pull/99/files/3c2da303f9a2239dad01857a4d861c5c81b21abe#r64094645>
>